CMake: miscellaneous fixes

MaartenBent opened the pull request:
Fix issues reported in https://trac.wxwidgets.org/ticket/18438.
Fix a warning when using `wxUSE_NO_RTTI`.
Set `wxUSE_DETECT_SM` off by default (same as configure). Especially because CMake does not have a check to detect `libSM`.

MaartenBent committed:
CMake: Disable finding X11 session manager by default

This is the same behaviour as configure has.

MaartenBent committed:
CMake: Only use -fno-rtti for C++

Using it for C generates the following warning:
command line option '-fno-rtti' is valid for C++/D/ObjC++ but not for C

MaartenBent committed:
CMake: Fix MSVC warning when using NMake

Fix warning: Command line warning D9025 : overriding '/W3' with '/W4'
Remove /W3, either via a regex or via new policy CMP0092 (CMake 3.15).

The policy has to be set before calling project(), otherwise the
CMAKE_<LANG>_FLAGS are already initialized.

See https://trac.wxwidgets.org/ticket/18438

MaartenBent committed:
CMake: Fill in default configuration in cmake-gui

The Debug configuration is the default, but it is not filled in in cmake-gui.
Do this, and set Debug and Release as available options.

See https://trac.wxwidgets.org/ticket/18438

vadz commented:
I wanted to merge this, but it looks like the last commit (18246ea) shouldn't be really merged, should it?

MaartenBent commented:
No. CMake on macOS fails for some unknown reason.

MaartenBent committed:
CMake: Fix detecting AppleClang

Policies are now correctly initialized (before creating the project).
Policy CMP0025 changes the compilerID of Clang on Apple, handle this correctly.
